sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/IDigitalTwinRelationship.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.digitaltwins.core;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Interface that defines the Json properties that are present in every digital twin relationship. Implementations of
+ * this class must add Json properties for each of the getters defined in this interface. See each getter for more
+ * details on each Json property.
+ */
+public interface IDigitalTwinRelationship {
+    public static String DIGITAL_TWIN_RELATIONSHIP_ID_JSON_PROPERTY_NAME = "$relationshipId";
+    public static String DIGITAL_TWIN_RELATIONSHIP_SOURCE_ID_JSON_PROPERTY_NAME = "$sourceId";
+    public static String DIGITAL_TWIN_RELATIONSHIP_TARGET_ID_JSON_PROPERTY_NAME = "$targetId";
+    public static String DIGITAL_TWIN_RELATIONSHIP_NAME_JSON_PROPERTY_NAME = "$relationshipName";
+
+    /**
+     * Gets the unique Id of the relationship. This field is present on every relationship.
+     * <p>
+     * The Json property associated with this Id is {@link IDigitalTwinRelationship#DIGITAL_TWIN_RELATIONSHIP_ID_JSON_PROPERTY_NAME},
+     * so every implementation of this class must have a Json property with that value. As an example, see
+     * {@link com.azure.digitaltwins.core.models.BasicRelationship} which demonstrates how to create this
+     * property using {@link JsonProperty}.
+     *
+     * @return The unique Id of the relationship. This field is present on every relationship.
+     */
+    public String getRelationshipId();
+
+    /**
+     * Gets the unique Id of the source digital twin. This field is present on every relationship.
+     * <p>
+     * The Json property associated with this source digital twin Id is
+     * {@link IDigitalTwinRelationship#DIGITAL_TWIN_RELATIONSHIP_SOURCE_ID_JSON_PROPERTY_NAME},
+     * so every implementation of this class must have a Json property with that value. As an example, see
+     * {@link com.azure.digitaltwins.core.models.BasicRelationship} which demonstrates how to create this
+     * property using {@link JsonProperty}.
+     *
+     * @return The unique Id of the source digital twin. This field is present on every relationship.
+     */
+    public String getSourceDigitalTwinId();
```

Comment:
In .NET this is just `SourceId`. I can't recall if there was Azure SDK board feedback to change the name here, or just a comment.

Reply:
We did get feedback to change this name to be more descriptive, yes

Reply:
From Srikanta:

> To disambiguate between the relationship id and the digital twin id, we should consider changing the names of getId() to getRelationshipId(), getSourceId() to getSourceDigitalTwinId() and getTargetId() to getTargetDigitalTwinId().